bokehjs/src/coffee/models/plots/plot.coffee

```diff
@@ -548,29 +548,56 @@ class PlotView extends Renderer.View
   resize: () =>
     @resize_width_height(true, false)
 
-  resize_width_height: (use_width, use_height, maintain_ar=true) =>
+  resize_width_height: (use_width, use_height, maintain_ar=true, width=null, height=null) =>
     # Resize plot based on available width and/or height
-
+    
+    # If size is explicitly given, we don't have to measure any DOM elements. Shortcut for Phosphor.
+    if typeof width is 'number' and typeof height is 'number'
+      return @_resize_width_height(use_width, use_height, maintain_ar, width, height)
+    
     # the solver falls over if we try and resize too small.
     # min_size is currently set in defaults to 120, we can make this
     # user-configurable in the future, as it may not be the right number
     # if people set a large border on their plots, for example.
 
-    # We need the parent node, because the el node itself collapses to zero
-    # height. It might not be available though, if the initial resize
+    # Try to find bk-root. We will query the parent of that node for its size
+    node = @.el
+    for i in [0..2]  # Use for-loop; if we ever change DOM structure, it should still work
+      if node is null or node.classList.contains('bk-root')
+         break
+      node = node.parentNode
+
+    # The plot node might be an orphan if the initial resize
     # happened before the plot was added to the DOM. If that happens, we
     # try again in increasingly larger intervals (the first try should just
     # work, but lets play it safe).
     @_re_resized = @_re_resized or 0
-    if not @.el.parentNode and @_re_resized < 14  # 2**14 ~ 16s
+    if not (node? and node.parentNode?) and @_re_resized < 14  # 2**14 ~ 16s
       setTimeout( (=> this.resize_width_height(use_width, use_height, maintain_ar)), 2**@_re_resized)
       @_re_resized += 1
       return
 
-    avail_width = @.el.clientWidth
-    avail_height = @.el.parentNode.clientHeight - 50  # -50 for x ticks
+    # Check that what we found is a bk-root. If not, this is probably a subplot, which we 
+    # can not currently make responsive in a good way
+    if not node.classList.contains('bk-root')
+       Bokeh.logger.warn('subplots cannot be responsive')
```

Comment:
You can't use `Bokeh` global in library code. There should be global `logger` in this module already.